RadixWallet/Features/TransactionReviewFeature/TransactionReview.swift

```diff
@@ -584,22 +584,33 @@ extension ResourceBalance {
 				liquidStakeUnit.guarantee
 			case let .poolUnit(poolUnit):
 				poolUnit.guarantee
-			case .nonFungible, .stakeClaimNFT:
+			case .nonFungible, .stakeClaimNFT, .none:
 				nil
 			}
 		}
 		set {
-			switch details {
-			case var .fungible(fungible):
-				fungible.guarantee = newValue
-				details = .fungible(fungible)
-			case var .liquidStakeUnit(liquidStakeUnit):
-				liquidStakeUnit.guarantee = newValue
-				details = .liquidStakeUnit(liquidStakeUnit)
-			case var .poolUnit(poolUnit):
-				poolUnit.guarantee = newValue
-				details = .poolUnit(poolUnit)
-			case .nonFungible, .stakeClaimNFT:
+			switch self {
+			case let .known(knownResourceBalance):
+				switch knownResourceBalance.details {
+				case var .fungible(fungible):
+					fungible.guarantee = newValue
+					var known = knownResourceBalance
+					known.details = .fungible(fungible)
+					self = .known(known)
+				case var .liquidStakeUnit(liquidStakeUnit):
+					liquidStakeUnit.guarantee = newValue
+					var known = knownResourceBalance
+					known.details = .liquidStakeUnit(liquidStakeUnit)
+					self = .known(known)
+				case var .poolUnit(poolUnit):
+					poolUnit.guarantee = newValue
+					var known = knownResourceBalance
+					known.details = .poolUnit(poolUnit)
+					self = .known(known)
+				case .nonFungible, .stakeClaimNFT:
+					return
```

Comment:
feels wrong to NOOP the `newValue` in case current value was `.nonFungible, . stakeClaimNFT, . unknown`, I Know we used to do it (before your changes) for `.nonFungible, . stakeClaimNFT`. But it looks wrong.

Can we at least add an `assertionFailure`?